service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -229,10 +229,19 @@ func (d *Decoder) decodeBinary(b []byte, v reflect.Value) error {
 			v.Index(i).SetUint(uint64(b[i]))
 		}
 	default:
-		if v.Kind() == reflect.Array && v.Type().Elem().Kind() == reflect.Uint8 {
-			reflect.Copy(v, reflect.ValueOf(b))
+		if v.Kind() == reflect.Array {
+			switch v.Type().Elem().Kind() {
+			case reflect.Uint8:
+				reflect.Copy(v, reflect.ValueOf(b))
+			case reflect.String:
+				v.SetString(string(b))
```

Comment:
I'm not sure this case is correct. If `v` is of type `[3]string`  I think `SetString` on `v` should panic. Should this be a copy per rune into `v`'s array, or a set of `v`'s first index?  Is `SetString` doing a copy? I didn't see that documented in the reflect package.